src/bun.js/api/bun/subprocess.zig

```diff
@@ -1802,6 +1845,41 @@ pub fn getExited(
     }
 }
 
+pub fn getSuccess(
+    this: *Subprocess,
+    this_value: JSValue,
+    globalThis: *JSGlobalObject,
+) JSValue {
+    if (JSC.Codegen.JSSubprocess.successPromiseGetCached(this_value)) |promise| {
+        return promise;
+    }
+
+    switch (this.process.status) {
```

Comment:
Ideally each clause should return an exit code and there should be unified code to resolve/reject `promise`. I was unable to get this to work and could use syntax advice.

```suggestion
    const exitCode: ?u8 = switch (this.process.status) {
```